### PR TITLE
feat(ONYX-380): add colours filter to saved searches

### PR DESCRIPTION
**Pull request by MounirDhahri**
This PR resolves [ONYX-380] 

### Description
This PR adds color filtering to saved searches filter

https://github.com/artsy/eigen/assets/11945712/73b1144c-f110-4b7e-a5a2-6cb9ef55b149

### PR Checklist

- [x] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- Add colours filtering to saved searches - mounir

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[ONYX-380]: https://artsyproduct.atlassian.net/browse/ONYX-380?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Cross-platform user-facing changes (Add colours filtering to saved searches - mounir)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 4c8a5ac70f768eec18fec780ad09829afc72f0f8
</p>

**Review by olerichter00 — APPROVED** [review]
Looks good 🌟

File: src/app/Scenes/SavedSearchAlert/Components/SavedSearchFilterColour.tsx [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+  COLORS_INDEXED_BY_VALUE,
+  COLORS_OPTIONS,
+  SWATCHES_PER_ROW,
+} from "app/Components/ArtworkFilter/Filters/ColorsOptions"
+import { ColorsSwatch } from "app/Components/ArtworkFilter/Filters/ColorsSwatch"
+import { SearchCriteria } from "app/Components/ArtworkFilter/SavedSearch/types"
+import { SavedSearchStore } from "app/Scenes/SavedSearchAlert/SavedSearchStore"
+import { isValueSelected, useSearchCriteriaAttributes } from "app/Scenes/SavedSearchAlert/helpers"
+
+export const SavedSearchFilterColour = () => {
```
[the remark this answers is not in the document]
Comment (MounirDhahri) [reply src/app/Scenes/SavedSearchAlert/Components/SavedSearchFilterColour.tsx:12]: lol - did it intentionally here to follow Barney's designs 😄

File: src/app/Scenes/SavedSearchAlert/Components/SavedSearchFilterColour.tsx [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    }
+  }
+
+  return (
+    <Flex>
+      <Text px={2} variant="sm" fontWeight={500}>
+        Colour
+      </Text>
+      <Spacer y={1} />
+      <Flex flexDirection="row" flexWrap="wrap" px={1}>
```
[the remark this answers is not in the document]
Comment (MounirDhahri) [reply src/app/Scenes/SavedSearchAlert/Components/SavedSearchFilterColour.tsx:51]: good point, will update

**Pull request merged**